game/src/ui/gameplay/HealthBar/HealthBar.gd

```diff
@@ -0,0 +1,60 @@
+extends Node2D
+
+
+export var init_red_portal_health: int = 100
+export var init_blue_portal_health: int = 100
+onready var red_portal_health: int = init_red_portal_health
+onready var blue_portal_health: int = init_blue_portal_health
+var red_health_display: TextureProgress
+var blue_health_display: TextureProgress
+
+func _ready():
+	red_health_display = self.get_node("Red")
+	blue_health_display = self.get_node("Blue")
+	red_health_display.set_max(init_red_portal_health)
```

Comment:
## Current (if i get you correctly)

- store two vars per colour: init health value and a health value
- set the maximum value of the health bar to be the init value
- let losses in health decrease the health var, and in turn decrease the fill of the bar

So init health var is only useful to set the max value (health X out Y, this sets the Y value).

E.g in rounds where you start with different health like tutorial starts with lets say 50, the bar is full at 50.

## Question:

Doesn't that defeat the purpose of the health bar if its not intuitive that you're starting at a low level? If you are starting at 50, i feel the bar should be half filled, the user should not have to notice the number value next to it to identify the weak portal.

Another way would be to set the max value always at 100, then the current health will always be the numerator over 100. (If we are setting 9999 as the health value, maybe can just fill the bar and put the infinity sign). In that scenario we wont' need the init health variable anymore and can remove it altogether.